include/openPMD/RecordComponent.hpp

```diff
@@ -123,17 +141,13 @@ RecordComponent::loadChunk(Offset const& o, Extent const& e, std::unique_ptr< T[
                                      + " - DS: " + std::to_string(dse[i])
                                      + " - Chunk: " + std::to_string(o[i] + e[i])
                                      + ")");
-    if( Allocation::API == alloc && data )
-        throw std::runtime_error("Preallocated pointer passed with signaled API-allocation during chunk loading.");
-    else if( Allocation::USER == alloc && !data )
-        throw std::runtime_error("Unallocated pointer passed with signaled user-allocation during chunk loading.");
+    if( !data )
+        throw std::runtime_error("Unallocated pointer passed during chunk loading.");
 
     uint64_t numPoints = 1u;
```

Comment:
This could now be pulled into a lower scope inside the ```if( *m_isConstant )``` branch.

Reply:
Lines https://github.com/ax3l/openPMD-api/blob/fd7b2b1b14e8b574c3b35d5ef00bb50f59441926/include/openPMD/RecordComponent.hpp#L176-L226 can now safely be removed as the functionality is provided now.